/*!
* @file
* @brief IDetector
*
* @copyright Copyright (C) University of Bristol 2017
*    This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. 
*    If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
*    See LICENSE file for details.
* @date 01/02/2018
* @author Richard Collins <richard.collins@bristol.ac.uk>
*/

syntax = "proto3";
import public "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

package cqp.remote;

/**
 * @brief The ISift class
 * Interface for the communication between Sifting classes
 */
service IDetector {
    /**
    * @brief StartDetecting
    * Instructs the detector to expect photons
    * @param request The timestamp for the start of the session
    * @return status
    */
    rpc StartDetecting(google.protobuf.Timestamp) returns (google.protobuf.Empty);

    /**
    * @brief StopDetecting
    * Instructs the detector that no more photons will be sent
    * @param request The timestamp for the end of the session
    * @return status
    */
    rpc StopDetecting(google.protobuf.Timestamp) returns (google.protobuf.Empty);
}
